### PR TITLE
revisited submit and missing form data

### PR DESCRIPTION
**Pull request by reepush**
New _doSubmit_ implementation. Now we focus on buttons for submitting forms with more flexible detection by adding `ACCEPT_PATTERNS` and `IGNORE_PATTERNS`. Also there is a fix for missing post requests from previous pull request.

Tested on the recently reported sites:
* https://web.tresorit.com/login
* https://www.vodafone.de/ussa/login/login.ftel
* https://nordinary-com.netcup-mail.de/SOGo/
* https://accounts.google.com/ServiceLogin

Also setting tests now.

### CODE REVIEW

File: chrome_extension/vendor/mooltipass/mcCombinations.js [hunk 1]
```diff
@@ -931,61 +931,40 @@ mcCombinations.prototype.doSubmit = function doSubmit( currentForm ) {
...
-		} else if ( mpJQ('#verify_user_btn').length > 0 ) { // Exclusive else/if for Autodesk.com (probably it could be used in more 2 steps login procedures)
-			mpJQ('#verify_user_btn').click();
-		} else if (window.location.hostname == 'accounts.google.com') { // Special case for google
-            if (mpJQ('#identifierNext').length > 0)
-            {
-                mpJQ('#identifierNext').click();
-            }
-            if (mpJQ('#passwordNext').length > 0)
-            {
-                mpJQ('#passwordNext').click();
```
Comment (limpkin) [inline chrome_extension/vendor/mooltipass/mcCombinations.js]: you know that this code is for google, right? 
any reason why you're simply removing it?

File: chrome_extension/vendor/mooltipass/mcCombinations.js [hunk 2]
```diff
@@ -931,61 +931,40 @@ mcCombinations.prototype.doSubmit = function doSubmit( currentForm ) {
...
+		})[0]
+		
+		$root = $($root.parent()[0] || mpJQ('body'))
+	}
+	
+	if (submitButton) {
+		// Select innermost element to trigger click. Event will be propagated anyway.
+		submitButton = mpJQ(submitButton).find(':not(:has(*))')[0] || submitButton
+		
+		mpJQ(submitButton).trigger('click')
```
Comment (limpkin) [inline chrome_extension/vendor/mooltipass/mcCombinations.js]: so before we were submitting and then clicking the button, and now we're doing the opposite? coud you explain why?

**Comment by limpkin** [comment]
how did you test on vodafone.de & nordinary when you can't create accounts there?

File: chrome_extension/vendor/mooltipass/mcCombinations.js (hunk 2 above)
Comment (limpkin) [inline chrome_extension/vendor/mooltipass/mcCombinations.js]: I should add that there's a very good reason why we used to do the way we did... but i'll monitor your CI tests very closely.

File: chrome_extension/vendor/mooltipass/mcCombinations.js (hunk 1 above)
Comment (reepush) [inline chrome_extension/vendor/mooltipass/mcCombinations.js]: it's handled by `ACCEPT_PATTERNS ` now, tested on google and it works

File: chrome_extension/vendor/mooltipass/mcCombinations.js (hunk 2 above)
Comment (reepush) [inline chrome_extension/vendor/mooltipass/mcCombinations.js]: All recently sites with issues have a problem with submitting default form without triggering expected handler by clicking on buttons. That's why I decided to rely on buttons first and then on submit handlers.

**Comment by reepush** [comment]
@limpkin 
We save credentials on form submit, even if it's not successful. And then refreshing the page I discovered the problem that expected submit handlers aren't triggered, because submit event isn't handled by scripts.

**Comment by reepush** [comment]
@limpkin
[Latest test results:](https://travis-ci.org/reepush/mooltipass/builds/253632000)

```
Supported.feature
18 Scenarios (1 failed, 17 passed)
```

Dropbox failed, but works on my local.

**Comment by limpkin** [comment]
@reepush yes, exactly like the master (dropbox problem is a layout one) ....

File: chrome_extension/background/init.js [hunk 3]
```diff
@@ -139,14 +139,11 @@ startMooltipass = function() {
...
-				// Deal both with RAW DATA and FORM DATA
-				// If we notice the extension being slow because of it intercepting too much data, uncomment the following line
-				//if (details && details.type === "xmlhttprequest") { // Raw data (multipart posts, etc)
-					if (details && details.requestBody && details.requestBody.raw && details.requestBody.raw[0]) {
-						var buffer = details.requestBody.raw[0].bytes;
-						var parsed = arrayBufferToData.toJSON(buffer);
-						if ( details.tabId ) chrome.tabs.sendMessage( details.tabId, {action: 'post_detected', post_data: parsed });	
-					//} 
+				// Deal with RAW DATA
+				if (details.requestBody && details.requestBody.raw && details.requestBody.raw[0]) {
```
Comment (limpkin) [inline chrome_extension/background/init.js]: could you keep the master version with the comments and check for "details"?

**Pull request merged**